autopush/endpoint.py

```diff
@@ -867,3 +868,67 @@ def _load_params(self):
             return json.loads(self.request.body)
         except ValueError:
             return {}
+
+
+class BridgeMessageHandler(RegistrationHandler):
+    cors_methods = "PUT"
+
+    @cyclone.web.asynchronous
+    def put(self, uaid):
+        """HTTP PUT
+
+        Process message states for a UAID
+
+        """
+        self.start_time = time.time()
+        if not self._validate_auth(uaid):
+            return self._write_unauthorized_response(
+                message="Invalid Authentication")
+        params = self._load_params()
+        if 'nak' in params:
+            d = deferToThread(self._record_naks, uaid, naks=params['nak'])
+        else:
+            d = Deferred()
+            d.callback(True)
+        d.addCallback(self._success)
+        d.addErrback(self._response_err)
+        return d
+
+    def _record_naks(self, o_uaid, naks):
+        """ Record the NAKs from the client
+
+        """
+        for nak in naks:
+            try:
+                result = self.ap_settings.fernet.decrypt(
+                    nak['id'].encode('utf8'))
+                kind, uaid, chid = result.split(":")
+                if kind != 'm':
+                    self.log.error(
```

Comment:
Since this value is usually echoed back from data provided in the subscription update, and it's encrypted and decrypted by our server, it's reasonable to presume it may have come from our server. Of course, there is the potential that it came from a client error, but since we have a tight relationship with the client code, it's worth noting.